src/common/commandExecutor.ts

```diff
@@ -96,33 +94,35 @@ export class CommandExecutor {
             }
         });
 
-        Q.delay(300).then(() => deferred.resolve(result.spawnedProcess));
+        // TODO - PROMISE: We need to consume result.outcome here
+        Q.delay(300).done(() => deferred.resolve(result.spawnedProcess));
         return deferred.promise;
     }
 
     /**
      * Kills the React Native packager in a child process.
      */
-    public killReactPackager(packagerProcess: ChildProcess, outputChannel?: OutputChannel): void {
+    public killReactPackager(packagerProcess: ChildProcess, outputChannel?: OutputChannel): Q.Promise<void> {
         Log.logMessage("Stopping Packager", outputChannel);
 
         if (packagerProcess) {
             /* To reliably kill the child process on all versions of Windows,
              * please use taskkill to end the packager process */
+            let waitForKill: Q.Promise<void>;
             if (process.platform === "win32") {
-                new Node.ChildProcess().exec("taskkill /pid " + packagerProcess.pid + " /T /F").outcome.then(() => {
+                waitForKill = new Node.ChildProcess().exec("taskkill /pid " + packagerProcess.pid + " /T /F").outcome.then(() => {
                     Log.logMessage("Packager stopped", outputChannel);
-                }, function() {
-                    Log.logError("Failed to exit the React Native packager", outputChannel);
                 });
             } else {
+                waitForKill = Q.resolve<void>(void 0);
                 packagerProcess.kill();
                 Log.logMessage("Packager stopped", outputChannel);
             }
 
-            packagerProcess = null;
+            return waitForKill.then(() => packagerProcess = null);
```

Comment:
This assignment is irrelevant / does nothing, since it is just an argument passed in to the function. This function can just return a promise in each of the branches.